app/src/main/java/com/example/criminalintent/MainActivity.kt

```diff
@@ -7,5 +7,14 @@ class MainActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
+        val currentFragment = supportFragmentManager.findFragmentById(R.id.fragment_container)
+        currentFragment.let {
```

Comment:
You can add a `if (currentFragment == null)` check here instead. `supportFragmentManager` will return the fragment if it's already in the list, for example when the activity is recreated after a screen rotation. In that case there's no need to create a new `CrimeFragment`